apps/aecore/lib/aecore/structures/signed_tx.ex

```diff
@@ -43,4 +43,31 @@ defmodule Aecore.Structures.SignedTx do
     end
   end
 
+  @spec is_oracle_query_tx(map()) :: boolean()
+  def is_oracle_query_tx(tx) do
+    Map.has_key?(tx, "sender") && Map.has_key?(tx, "oracle_hash") &&
```

Comment:
wouldn't pattern match tx.data type be better?

Reply:
ah I didn't noticed the serialization uses it for both directions, so it has to be like this